Clinical-Genomics/microSALT
# Auxillary Quality of Life

# Description

### End-user:
* Printing (apple-p) a report now shows typical info (file name, page number)
* QC & ST report names now end in version number

### Bioinformaticians:
* Restructured subdirectories:
  * Removed 'trimmed' directory
  * 'resistance' and 'blast' now go under 'blast_search/resistance' and 'blast_search/mlst'
  * 'quast' is now a subdirectory of assembly
* NovaSeq run dates now display properly
* Added file `config.log` in each result directory which describes stores the configuration used to analyse
* Installation is now a 1-liner

### Internals:
* Expanded special report 'resistance overview' with 'motif overview' (which contain EXPAC virulence). The references for this is provided by microSALT.
* Improved special report 'ST Updates' to swedish translation and improved look
* All conda dependencies (blast, bwa etc) updated to latest hasta-compatible versions
* Flask webserver delay increased to reduce (unlikely) chance of reports being generated _too_ fast
* Incorrect installation warning (quast) resolved
* Removed adapters path from microSALT config, integrated into code

## Primary function of PR
- [X] New type of analysis (Outside regular scope)

# Testing

- `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh aux_qol`
- `us`
- `source activate S_microSALT`
- `microSALT analyse project MIC3109`
- `microSALT analyse project MIC4107`
- `microSALT analyse project MIC4109`
- `microSALT analyse project ACC5702`

1. Review the MIC3109 QC and MIC4107 & MIC4109 Typing reports. They should not deviate from the values presented in document 1490. 
2. Review a subset of samples of ACC5702. The values 'Ankomstdatum' and 'Datum sekvensering' should exist and correlate what is found in the LIMS.
3. Check the results folder for a few of the samples. Make sure the organization blast_search/mlst, blast_search/resistance, assembly/quast and lack of 'trimmed' directory (see above) holds true.
4. Find the config.log file for the samples in step three (see above). Make sure it exists and the information within it mimics the configuration used.
5. Run `microSALT utils report XXX --type motif_overview` to generate an EXPAC report. Make sure the report looks sensible.
6. Run `microSALT utils resync review XXX --type report` to generate an ST update report. Make sure the report looks sensible.

## Test results
1. Coverage for MIC3109 deviated due to version changes. No individual value deviated more than 0.5% compared to the original, which is acceptable according to 1490. As a result of this, document 1490 will have it's values replaced.
2. Outputted as expected.
3. Looks just as described.
4. Looks just as described.
5. Report looks similar to previous resistance gene report.
6. Nothing looks obviously wrong with the ST report.

# Sign-offs
- [x] Approved to run at Clinical-Genomics by @patrikgrenfeldt
